### ISSUE TITLE
Simplify UX to Enable VCP - Manual testing

### ISSUE THREAD

**Issue by divyenpatel** [vmware/kubernetes#215]
Testing and bug fixes.

**Comment by tusharnt** [comment]
Load balancing. @BaluDontu, please check with @divyenpatel on what needs to be done here.

**Comment by divyenpatel** [comment]
New bootstrap ignition image is ready to deploy Kubernetes Cluster with vSphere Cloud Provider disabled: `docker.io/divyen/k8s-ignition:disabledvcp`

I will post testing steps here.

**Comment by divyenpatel** [comment]
@rohitjogvmw Here are the steps for testing:


Deploy Kubernetes Cluster with vSphere Cloud Provider Disabled.

```
* Phase 2: Node Bootstrapping
*
kubernetes version (phase2.kubernetes_version) [v1.6.5] (NEW) 
bootstrap provider (phase2.provider) [ignition] (NEW) 
  installer container (phase2.installer_container) [docker.io/cnastorage/k8s-ignition:v2] (NEW) docker.io/divyen/k8s-ignition:disabledvcp
  docker registry (phase2.docker_registry) [gcr.io/google-containers] (NEW) 
*
```

Once Cluster is up.

Download https://github.com/vmware/kubernetes/blob/enable-vcp-uxi-fixes/enable-vsphere-cloud-provider.yaml , fill up configuration details. and kick off configuration pods.

```
$ kubectl create -f enable-vsphere-cloud-provider.yaml
namespace "vmware" created
secret "vsphere-cloud-provider-secret" created
serviceaccount "vcpsa" created
pod "vcp-manager" created
```

Check out VcpSummary status at regular interval. when you see `nodes_sucessfully_configured` equals the `total_number_of_nodes`, vSphere Cloud Provider is successfully configured.
 
```
kubectl get VcpSummary --namespace=vmware -o json
.
.
 "nodes_sucessfully_configured": "5",
 "total_number_of_nodes": "5"
```


Logs of individual daemonset pods can be retrieved using

```
$ kubectl get pods --namespace=vmware
NAME                   READY     STATUS    RESTARTS   AGE
vcp-daementset-1xf2b   1/1       Running   0          9m
vcp-daementset-3qzmv   1/1       Running   0          9m
vcp-daementset-nvczt   1/1       Running   0          9m
vcp-daementset-s1tl6   1/1       Running   0          9m
vcp-daementset-wtcnx   1/1       Running   0          9m
vcp-manager            1/1       Running   0          9m

$ kubectl logs vcp-daementset-1xf2b --namespace=vmware
Running script in the Pod: vcp-daementset-1xf2b deployed on the Node: kubernetes-node4
Running Daemon Role
error: unable to recognize "/tmp/vcp-daementset-1xf2b_daemonset_status_create.yaml": no matches for vmware.com/, Kind=VcpStatus
/opt/enable-vcp-scripts/common_func.sh: line 213: 1: command not found
/opt/enable-vcp-scripts/common_func.sh: line 213: 2: command not found
Running script in the Pod: vcp-daementset-1xf2b deployed on the Node: kubernetes-node4
[INFO] Successfully enabled disk.enableUUID flag on the Node Virtual Machine.
[INFO] Moved Node Virtual Machine to the Working Directory Folder.
[INFO] Creating directory: '/host/tmp/vcp-daementset-1xf2b' for back up of manifest files and kubelet service configuration file
[INFO] Successfully created back up directory: /host/tmp/vcp-daementset-1xf2b  on kubernetes-node4 node
[INFO] Verified that the directory for the vSphere Cloud Provider configuration file is accessible. Path: (/host//etc/kubernetes )
[INFO] vsphere.conf file is already available at /host//etc/kubernetes/vsphere.conf
[INFO] Existing vsphere.conf file is copied to /host/tmp/vcp-daementset-1xf2b/vsphere.conf
[INFO] Found file: /host//etc/systemd/system/kubelet.service
[INFO] Successfully backed up  /host//etc/systemd/system/kubelet.service at /host/tmp/vcp-daementset-1xf2b
[INFO] Creating vSphere Cloud Provider configuration file at /host/tmp/vsphere.conf
[INFO] successfully created vSphere.conf file at : /host/tmp/vsphere.conf
[INFO] Found file: /host//etc/systemd/system/kubelet.service
[INFO] Sucessfully updated kubelet.service configuration
[INFO] Reloading systemd manager configuration and restarting kubelet service
[INFO] kubelet service restarted sucessfully
```

**Comment by divyenpatel** [comment]
Observed couple of Issues(https://github.com/vmware/kubernetes/issues/259, https://github.com/vmware/kubernetes/issues/257) during 2nd round of testing with RBAC enabled. and fixed them as part of https://github.com/vmware/kubernetes/commit/89d61c732daec34a66fe2c3f950e5a7b510ac0b0

**Comment by divyenpatel** [comment]
@rohitjogvmw Instruction is posted at https://github.com/vmware/kubernetes/blob/enable-vcp-uxi/README.md

Deploy kubernetes cluster using https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/ or kubernetes-anywhere using the ignition image - `divyen/k8s-ignition:disabledvcp`
and follow instructions link mentioned above.

**Comment by rohitjogvmw** [comment]
I tested the steps with below setup details

> k8s cluster running on vSphere.
> All k8s nodes (including master) were running in same ESXi cluster

I found below missing steps:
 1. highlight about taint node so that daemonset can be created

 2. default datastore: Update node specifying requirement of shared datastore

 3. Provide instruction for "how to get kubelet instruction file"
     May be with example it will be simple to understand
     Also please provide details how to use this instruction file.

4. kubenetes has 2 files: kubelet.conf and admin.conf. admin.conf need to be  copied in ~/.kube/config so that kubectl gets full access to the k8s cluster

Once above steps are fixed, I shall retest with more cases.
Note: Above setup is happy-path setup. I will trying with different types of k8s deployments like k8s spanning across multiple clusters, datacenters etc. to see if script correctly fails or not.

**Comment by rohitjogvmw** [comment]
Tested with the changes done by Divyen.
I was able to enable VCP on k8s cluster.

Tested below:
k8s version: 1.7.6
vSphere version: 6.5

Deployed k8s cluster in single DC and tried enabling VCP.
Officially multiple DC, multiple VC etc. are not supported so did not checked that.

Even though  VCP can work in k8s cluster spanning across multiple ESXi clusters within one DC, as secrete.yaml asks for only one cluster-name, did not check that case. Whenever UX scripts adds support for that, I can test it again.
